src/main/app/src/Components/Landscape/Search/Search.tsx

```diff
@@ -31,39 +43,50 @@ interface ILabelValue {
   value: number;
 }
 
-const Search: React.FC<PropsInterface> = (props: PropsInterface) => {
+const Search: React.FC<PropsInterface> = ({findFunction, setSidebarContent, ...props}) => {
   const match: { params?: { identifier?: string } } | null = matchPath(props.location.pathname, {
     path: Routes.MAP_ROUTE,
     exact: false,
     strict: false,
   });
 
   const identifier = match?.params?.identifier;
+  const [currentLandscape, setCurrentLandscape] = useState<string>('');
   const [results, setResults] = useState<IItem[]>([]);
   const [facets, setFacets] = useState<IFacet[]>([]);
   const [searchTerm, setSearchTerm] = useState('');
   const [hasChange, setHasChange] = useState(false);
-  const [hasFocus, setHasFocus] = useState(false);
-
+  const classes = useStyles();
+  const componentClasses = componentStyles();
   const searchInput = React.useRef<HTMLDivElement>(null);
 
-  async function search(searchTerm: string, identifier: string) {
-    if (searchTerm.length < 2) return;
-
-    const result: IItem[] | null = await get(
-      '/api/landscape/' +
-        identifier +
-        '/search/' +
-        encodeURIComponent(searchTerm)
-          .replace(/[!'()]/g, escape)
-          .replace(/\*/g, '%2A')
-    );
-
-    if (!result) return;
-
-    setResults(result);
-    setHasChange(false);
-  }
+  const search = useCallback(
+    (searchTerm: string, identifier: string) => {
+      if (searchTerm.length < 2) return;
+
+      get(
+        '/api/landscape/' +
+          identifier +
+          '/search/' +
+          encodeURIComponent(searchTerm)
+            .replace(/[!'()]/g, escape)
+            .replace(/\*/g, '%2A')
```

Comment:
I like the search field 👍 

Unfortunately, it does not work well all the time. This should be better (backend functionality):

- find results regardless of upper and lowercase input
- find also partial words, e.g. with "word" I would like to already get "wordpress". Maybe one could match once an user has typed 3 chars
- "emailer" does not find the emailer. This is most probably a bug?

Reply:
I've quoted your remarks in https://github.com/dedica-team/nivio/issues/302